README.md

```diff
@@ -351,10 +351,131 @@ practice for modules to claim their own area in the configuration object is to
 insert their default configuration object using a key that either matches their
 module name or that matches their module name minus any `bedrock-` prefix. For
 example, the [bedrock-server][] module's specific configuration object can be
-found under `bedrock.config.server`. Modules may define whatever configuration
+found under `bedrock.config.server`. A `megacorp-feature` module would be found
+under `bedrock['megacorp-feature']`. Modules may define whatever configuration
 variables they want to using whatever format is appropriate for their own
 use.
 
+The `bedrock.util` module has helpers to setup configurations, and in
+particular, dynamic computed configurations. Computed values can help to
+simplify dependency issues by allowing values to be computed at runtime from a
+function or string template. (Note there is a small cost with computed config
+values which could be important depending on the use case.)
+
+`bedrock.util.Config` creates a wrapper around a config object and optional
+options. A common setup could look like the following.
+
+```js
+// an empty config object
+let config = {};
+// common options
+let options = {
+  // the config
+  config: config
+  // local vars used during string template evaluation
+  locals: config
+};
+// wrap the config
+let c = new bedrock.util.Config(config, options);
+```
+
+Bedrock provides a shared wrapper around the common `bedrock.config` as
+`bedrock.util.config`.
+
+To do simple sets of config data, use the `set()` API.
+
+```js
+let c = bedrock.util.config;
+// set a config variable with a path
+// path components are created as needed
+c.set('server.port', 8443);
+// config is now {"server": {"port": 8443}}
+```
+
+Normal code and the config API can be mixed. A useful helper is `require()`.
```

Comment:
I think `init` implies you are fully initializing something vs just making sure it exists.  `init` could be done like set('foo.bar', {})